tests/test_hgexports.py

```diff
@@ -581,3 +594,261 @@ def test_check_commit_message_invalid_message(
+        repo=valid_repo,
+        commit_message=COMMIT_MESSAGE,
+    )
+    assert (
+        diff_assessor.check_wpt_sync() is None
+    ), "Check should pass when user is not `wptsync@mozilla.com`."
+
+
+def test_check_wpt_sync_no_repo():
+    parsed_diff = rs_parsepatch.get_diffs(
+        GIT_DIFF_FILENAME_TEMPLATE.format(filename="somefile.txt")
+    )
+    diff_assessor = DiffAssessor(
+        author="wptsync@mozilla.com",
+        parsed_diff=parsed_diff,
+        commit_message=COMMIT_MESSAGE,
+    )
+    assert (
+        diff_assessor.check_wpt_sync() is None
+    ), "Check should pass without repo information."
+
+
+def test_check_wpt_sync_try(mocked_repo_config):
+    supported_repos = get_repos_for_env("test")
+    try_repo = supported_repos["try"]
+
+    parsed_diff = rs_parsepatch.get_diffs(
+        GIT_DIFF_FILENAME_TEMPLATE.format(filename="somefile.txt")
+    )
+    diff_assessor = DiffAssessor(
+        author="wptsync@mozilla.com",
+        parsed_diff=parsed_diff,
+        commit_message=COMMIT_MESSAGE,
+        repo=try_repo,
+    )
+    assert diff_assessor.check_wpt_sync() is None, "Check should pass for try repo."
+
+
+def test_check_wpt_sync_non_central_repo(mocked_repo_config):
+    supported_repos = get_repos_for_env("test")
+    invalid_repo = supported_repos["mozilla-new"]
+
+    parsed_diff = rs_parsepatch.get_diffs(
+        GIT_DIFF_FILENAME_TEMPLATE.format(filename="somefile.txt")
+    )
+    diff_assessor = DiffAssessor(
+        author="wptsync@mozilla.com",
+        parsed_diff=parsed_diff,
+        commit_message=COMMIT_MESSAGE,
+        repo=invalid_repo,
+    )
+    assert (
+        diff_assessor.check_wpt_sync() == "WPT Sync bot can not push to mozilla-new."
+    ), "Check should fail if WPTSync bot pushes to disallowed repo."
+
+
+def test_check_wpt_sync_invalid_paths(mocked_repo_config):
+    supported_repos = get_repos_for_env("test")
+    valid_repo = supported_repos["mozilla-central"]
+
+    parsed_diff = rs_parsepatch.get_diffs(
+        GIT_DIFF_FILENAME_TEMPLATE.format(filename="somefile.txt")
+    )
+    diff_assessor = DiffAssessor(
+        author="wptsync@mozilla.com",
+        parsed_diff=parsed_diff,
+        commit_message=COMMIT_MESSAGE,
+        repo=valid_repo,
+    )
+    assert diff_assessor.check_wpt_sync() == (
+        "Revision allows WPTSync bot to make changes to disallowed "
+        "files `somefile.txt`."
+    ), "Check should fail if WPTSync bot pushes to disallowed repo."
+
+
+def test_check_wpt_sync_valid_paths(mocked_repo_config):
+    supported_repos = get_repos_for_env("test")
+    valid_repo = supported_repos["mozilla-central"]
+
+    parsed_diff = rs_parsepatch.get_diffs(
+        GIT_DIFF_FILENAME_TEMPLATE.format(filename="testing/web-platform/moz.build")
+    )
+    diff_assessor = DiffAssessor(
+        author="wptsync@mozilla.com",
+        parsed_diff=parsed_diff,
+        commit_message=COMMIT_MESSAGE,
+        repo=valid_repo,
+    )
```

Comment:
I think some if not most of this code could go into a fixture, it's repeated throughout different tests, with minor differences between them.

Reply:
I have plans to make a few changes to this API in the near future, to account for hooks which need to inspect the entire contents of the push. Once I make those changes in a later PR, if a test fixture still makes sense here I will update accordingly.

There is some repeated code here, but even if this was in a test fixture, the fixture would likely end up having to take most of the same arguments as the raw `DiffAssessor` object.